src/contexts/movie-context.tsx

```diff
@@ -0,0 +1,73 @@
+import React, {
+  Dispatch,
+  FC,
+  ReactNode,
+  SetStateAction,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+import movie1 from '../assets/Movie1.jpg';
+
+const defaultMovies = Array(6).fill({
+  img: movie1,
+  title: 'Pulp Fiction',
+  year: 2004,
+  rating: 4.3,
+  duration: '154 min',
+  shortDescription: 'Action & Adventure',
+  description:
+    'Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.—Soumitra',
+});
+export type Movie = {
+  img: string;
+  title: string;
+  year: number;
+  rating: number;
+  duration: string;
+  shortDescription: string;
+  description: string;
+};
+interface ContextValue {
+  movies: Movie[];
+  selectedMovie?: Movie;
+  setSelectedMovie: Dispatch<SetStateAction<Movie | undefined>>;
+}
+
+const MovieContext = React.createContext<ContextValue | undefined>(undefined);
+
+interface ProviderProps {
+  children: ReactNode;
```

Comment:
No need to use this interface in scope of `React.FC` since `children` is already presented here

Reply:
> No need to use this interface in scope of `React.FC` since `children` is already presented here

it's no longer the case in React18
https://stackoverflow.com/questions/71788254/react-18-typescript-children-fc#answer-71809927

Reply:
Understood, thank you for sharing this info! I didn't know about that